core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalStatsTests.java

```diff
@@ -63,7 +65,34 @@ protected void assertReduced(InternalStats reduced, List<InternalStats> inputs)
     }
 
     @Override
+    protected void assertFromXContent(InternalStats aggregation, ParsedAggregation parsedAggregation) {
+        assertTrue(parsedAggregation instanceof ParsedStats);
+        ParsedStats parsed = (ParsedStats) parsedAggregation;
+        long count = aggregation.getCount();
+        assertEquals(count, parsed.getCount());
+        // for count == 0, fields are rendered as `null`, so  we test that we parse to default values used also in the reduce phase
+        assertEquals(count > 0 ? aggregation.getMin() : Double.POSITIVE_INFINITY , parsed.getMin(), 0);
+        assertEquals(count > 0 ? aggregation.getMax() : Double.NEGATIVE_INFINITY, parsed.getMax(), 0);
+        assertEquals(count > 0 ? aggregation.getSum() : 0, parsed.getSum(), 0);
+        assertEquals(count > 0 ? aggregation.getAvg() : 0, parsed.getAvg(), 0);
+        // also as_string values are only rendered for count != 0
+        if (count > 0) {
+            assertEquals(aggregation.getMinAsString(), parsed.getMinAsString());
+            assertEquals(aggregation.getMaxAsString(), parsed.getMaxAsString());
+            assertEquals(aggregation.getSumAsString(), parsed.getSumAsString());
+            assertEquals(aggregation.getAvgAsString(), parsed.getAvgAsString());
+            // NORELEASE there is no COUNT_AS_STRING in the json output, so we cannot get back a formatted value here
```

Comment:
This is one caveat I found in the current InternalStats implementation. We dont have a COUNT_AS_STRING field in the xContent output, but the java API provides formatted String values of the "long" count via getCountAsString(). Possible solutions I can think of for now:
* add COUNT_AS_STRING to xContent (breaking ?)
* change getCountAsString to return Long.toString(count)
* remove getCountAsString() from the Stats interface (after all we don't render a formatted version)
* ignore this as a rare edge case and exclude it from testing like I've done it now

Wdyt?